diff -ur gdbm-1.8.0/Makefile.in gdbm-1.8.0-mingw32/Makefile.in
--- gdbm-1.8.0/Makefile.in      Wed May 19 05:16:06 1999
+++ gdbm-1.8.0-mingw32/Makefile.in      Wed Aug  8 20:05:47 2001
@@ -122,11 +122,9 @@
 install: libgdbm.la gdbm.h gdbm.info
        $(srcdir)/mkinstalldirs $(libdir) $(includedir) $(man3dir) $(infodir)
        $(LIBTOOL) install -c libgdbm.la $(libdir)/libgdbm.la
-       $(INSTALL_DATA) -o $(BINOWN) -g $(BINGRP) gdbm.h $(includedir)/gdbm.h
-       $(INSTALL_DATA) -o $(BINOWN) -g $(BINGRP) $(srcdir)/gdbm.3 \
-               $(man3dir)/gdbm.3
-       $(INSTALL_DATA) -o $(BINOWN) -g $(BINGRP) $(srcdir)/gdbm.info \
-               $(infodir)/gdbm.info
+       $(INSTALL_DATA) gdbm.h $(includedir)/gdbm.h
+       $(INSTALL_DATA) $(srcdir)/gdbm.3 $(man3dir)/gdbm.3
+       $(INSTALL_DATA) $(srcdir)/gdbm.info $(infodir)/gdbm.info

 install-compat:
        $(srcdir)/mkinstalldirs $(includedir)
diff -ur gdbm-1.8.0/systems.h gdbm-1.8.0-mingw32/systems.h
--- gdbm-1.8.0/systems.h        Wed May 19 05:09:46 1999
+++ gdbm-1.8.0-mingw32/systems.h        Mon Apr 30 12:55:34 2001
@@ -146,3 +146,27 @@
 #ifndef STDERR_FILENO
 #define STDERR_FILENO 2
 #endif
+
+       //      Added by "Dave Roth" rothd@roth.net 970829
+       //      Modifications made by "Sergei Romanenko" <roman@spp.Keldysh.ru> 980602
+
+#ifdef WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <io.h>
+int    lock_val;
+#undef UNLOCK_FILE
+#define        UNLOCK_FILE(dbf) lock_val = !UnlockFile((HANDLE) _get_osfhandle((int) dbf->desc ), 0, 0, 0xffffffff, 0xffffffff);
+#undef WRITELOCK_FILE
+#define        WRITELOCK_FILE(dbf)      lock_val = !LockFile((HANDLE) _get_osfhandle((int) dbf->desc ), 0, 0, 0xffffffff, 0xffffffff);
+#undef READLOCK_FILE
+#define        READLOCK_FILE(dbf)      lock_val = !LockFile((HANDLE) _get_osfhandle((int) dbf->desc ), 0, 0, 0xffffffff, 0xffffffff);
+
+#undef fsync
+#define fsync(dbf)  _commit(dbf)
+
+       //      Redefine open() so files are opened in binary mode
+#define        open(x, y, z)   open(x, y | O_BINARY, z)
+#define link(x, y) 0
+
+#endif //      WIN32
